modules/cudastereo/src/cuda/stereosgm.cu

```diff
@@ -0,0 +1,2087 @@
+                {
+                    left_dest(0, x) = uniq ? compute_disparity(bestDisp, bestCost, smem_cost_sum[warp_id][smem_x]) : INVALID_DISP;
+                }
+            }
+        }
+    }
+    for (unsigned int i = 0; i < REDUCTION_PER_THREAD; ++i)
+    {
+        const unsigned int k = lane_id * REDUCTION_PER_THREAD + i;
+        const int p = static_cast<int>(((width - k) & ~(MAX_DISPARITY - 1)) + k);
+        if (0 <= p && p < width)
+        {
+            right_dest(0, p) = compute_disparity_normal(unpack_index(right_best[i]));
+        }
+    }
+}
+} // anonymous namespace
+
+template <size_t MAX_DISPARITY>
+void winnerTakesAll(const GpuMat& src, GpuMat& left, GpuMat& right, float uniqueness, bool subpixel, int mode, cv::cuda::Stream& _stream)
+{
+    cv::Size size = left.size();
+    int num_paths = mode == StereoSGBM::MODE_HH4 ? 4 : 8;
+    CV_Assert(src.rows == 1 && src.cols == size.width * size.height * MAX_DISPARITY * num_paths);
+    CV_Assert(size == right.size());
+    CV_Assert(left.type() == right.type());
+    CV_Assert(src.type() == CV_8UC1);
+    CV_Assert(mode == StereoSGBM::MODE_HH || mode == StereoSGBM::MODE_HH4);
+    const int gdim =
+        (size.height + WARPS_PER_BLOCK - 1) / WARPS_PER_BLOCK;
+    const int bdim = BLOCK_SIZE;
+    cudaStream_t stream = cv::cuda::StreamAccessor::getStream(_stream);
+    if (subpixel && mode == StereoSGBM::MODE_HH)
+    {
+        winner_takes_all_kernel<MAX_DISPARITY, 8, compute_disparity_subpixel<MAX_DISPARITY>><<<gdim, bdim, 0, stream>>>(
+            src, left, right, size.width, size.height, uniqueness);
+    }
+    else if (subpixel && mode == StereoSGBM::MODE_HH4)
+    {
+        winner_takes_all_kernel<MAX_DISPARITY, 4, compute_disparity_subpixel<MAX_DISPARITY>><<<gdim, bdim, 0, stream>>>(
+            src, left, right, size.width, size.height, uniqueness);
+    }
+    else if (!subpixel && mode == StereoSGBM::MODE_HH)
+    {
+        winner_takes_all_kernel<MAX_DISPARITY, 8, compute_disparity_normal><<<gdim, bdim, 0, stream>>>(
+            src, left, right, size.width, size.height, uniqueness);
+    }
+    else /* if (!subpixel && mode == StereoSGBM::MODE_HH4) */
+    {
+        winner_takes_all_kernel<MAX_DISPARITY, 4, compute_disparity_normal><<<gdim, bdim, 0, stream>>>(
+            src, left, right, size.width, size.height, uniqueness);
+    }
+}
+
+template CV_EXPORTS void winnerTakesAll< 64>(const GpuMat&, GpuMat&, GpuMat&, float, bool, int, cv::cuda::Stream&);
+template CV_EXPORTS void winnerTakesAll<128>(const GpuMat&, GpuMat&, GpuMat&, float, bool, int, cv::cuda::Stream&);
+template CV_EXPORTS void winnerTakesAll<256>(const GpuMat&, GpuMat&, GpuMat&, float, bool, int, cv::cuda::Stream&);
+} // namespace winner_takes_all
+
+namespace median_filter
+{
+namespace
+{
+const int BLOCK_X = 16;
+const int BLOCK_Y = 16;
+const int KSIZE = 3;
+const int RADIUS = KSIZE / 2;
+const int KSIZE_SQ = KSIZE * KSIZE;
+
+inline int divup(int total, int grain)
```

Comment:
This function is already defined in [cudev/common.hpp](https://github.com/opencv/opencv/blob/master/modules/core/include/opencv2/core/cuda/common.hpp#L94).
You should use that one.